### PR TITLE
5 westport fixes name, indexing, offshore chunking

### PR DESCRIPTION
**Pull request by rosepearson**
This pull request is in response to several issues encountered when producing a DEM for Westport.
Changes include:
1. Name the created dem with the binning approach -> idw
2. Ensure the created dem has positively increasing x and y indexing
3. Apply a tiling approach to the offshore interpolation to limit the maximum requirement

For this PR a code review of changes is probably all that is required.

### CODE REVIEW

**Review by LukeParky — CHANGES_REQUESTED** [review]
Overall high quality code. Some of this is pushing me in domain knowledge now but I added some suggestions for simplifying code

File: src/geofabrics/dem.py [hunk 1]
```diff
@@ -164,9 +168,11 @@ def _set_up(self):
...
             dem_temp.load()
             dem_temp.rio.set_crs(self.catchment_geometry.crs)
-        if self.raster_origin[0] != dem_temp.x.data.min() or self.raster_origin[1] != dem_temp.y.data.min():
-            raster_origin = [dem_temp.x.data.min() - self.catchment_geometry.resolution/2,
-                             dem_temp.y.data.min() - self.catchment_geometry.resolution/2]
+
+        # check if the raster origin has been moved by PDAL writers.gdal
+        raster_origin = [dem_temp.x.data.min() - self.catchment_geometry.resolution/2,
+                         dem_temp.y.data.min() - self.catchment_geometry.resolution/2]
+        if self.raster_origin[0] != raster_origin[0] or self.raster_origin[1] != raster_origin[1]:
```
[the remark this answers is not in the document]
Comment (rosepearson) [reply src/geofabrics/dem.py:175]: Does that work for lists! I will try this now.

[the remark this answers is not in the document]
Comment (rosepearson) [reply src/geofabrics/dem.py]: Done - nice suggestion

File: src/geofabrics/dem.py [hunk 2]
```diff
@@ -242,8 +248,25 @@ def dem(self):
...
+
+        x = dem.x
+        y = dem.y
+        if x[0] > x[-1]:
+            x = x[::-1]
+        if y[0] > y[-1]:
+            y = y[::-1]
+        dem = dem.reindex(x=x, y=y)
+        return dem
+
```
[the remark this answers is not in the document]
Comment (rosepearson) [reply src/geofabrics/dem.py]: Done. Nice suggestion.

File: src/geofabrics/processor.py [hunk 3]
```diff
@@ -128,7 +133,8 @@ def run(self):
 
         # Load in reference DEM if any significant land/foreshore not covered by LiDAR
         if (self.catchment_geometry.land_and_foreshore_without_lidar(self.catchment_lidar.extents).geometry.area.sum()
-                > self.catchment_geometry.land_and_foreshore.area.sum() * area_threshold):
+                > self.catchment_geometry.land_and_foreshore.area.sum() * area_threshold and
+                self.check_instruction_path('reference_dems')):
```
[the remark this answers is not in the document]
Comment (rosepearson) [reply src/geofabrics/processor.py]: Ok - I've had a good. Reopen if still needs work!

**Comment by rosepearson** [comment]
> Overall high quality code. Some of this is pushing me in domain knowledge now but I added some suggestions for simplifying code

Thanks @LukeParky. Great suggestions as always. I have addressed them. Let me know what you think!

**Review by LukeParky — APPROVED** [review]

**Comment by LukeParky** [comment]
Thanks for those changes, I approve it, might be worth keeping it open for @AliceHarang to double check

**Comment by rosepearson** [comment]
> Thanks for those changes, I approve it, might be worth keeping it open for @AliceHarang to double check

Whoops - probably would have been. I was a bit keen.

**Pull request merged**